File: Challenges/Algorithms/Warmup/Compare the Triplets/compare-the-triplets.py
#!/bin/python3

import sys


def compare(a, b):
    score_player_a = 0
    score_player_b = 0

    if a > b:
        score_player_a += 1
    elif a < b:
        score_player_b += 1

    return score_player_a, score_player_b


def solve(a0, a1, a2, b0, b1, b2):
    first = compare(a0, b0)
    second = compare(a1, b1)
    third = compare(a2, b2)

    score_alice = first[0] + second[0] + third[0]
    score_bob = first[1] + second[1] + third[1]
    return str(score_alice), str(score_bob)


a0, a1, a2 = input().strip().split(' ')
a0, a1, a2 = [int(a0), int(a1), int(a2)]
b0, b1, b2 = input().strip().split(' ')
b0, b1, b2 = [int(b0), int(b1), int(b2)]
result = solve(a0, a1, a2, b0, b1, b2)
print (" ".join(map(str, result)))

File: Challenges/Algorithms/Warmup/Compare the Triplets/compare-the-triplets.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    int a0, a1, a2, b0, b1, b2;
    int score_alice = 0;
    int score_bob = 0;
    std::cin >> a0 >> a1 >> a2 >> b0 >> b1 >> b2;

    auto compare = [&score_alice, &score_bob](int a, int b) {
        if (a > b)
            score_alice += 1;
        else if (a < b)
            score_bob += 1;
    };

    compare(a0, b0);
    compare(a1, b1);
    compare(a2, b2);

    std::cout << std::to_string(score_alice) << " " << std::to_string(score_bob);
    return 0;
}
